circle.yml

```diff
@@ -36,3 +36,25 @@ jobs:
             if [ "${CIRCLE_BRANCH}" == "release" ]; then
               make release
             fi
+  publish:
+    machine:
+      image: ubuntu-1604:202007-01
+    working_directory: /home/circleci/logspout
+    environment:
+      DEBUG: true
+    steps:
+      - checkout
+      - run: mkdir -vp ~/.docker/cli-plugins/
+      - run: curl --silent -L --output ~/.docker/cli-plugins/docker-buildx https://github.com/docker/buildx/releases/download/v0.3.1/buildx-v0.3.1.linux-amd64
+      - run: chmod a+x ~/.docker/cli-plugins/docker-buildx
+      - run: docker login -u "$DOCKER_USERNAME" -p "$DOCKER_PASSWORD"
```

Comment:
how are `DOCKER_USERNAME` and `DOCKER_PASSWORD` defined?

Reply:
To set **DOCKER_USERNAME** and **DOCKER_PASSWORD**, we need to choose the **Environment Variables** option and set the username and password for the dockerhub in **CircleCI project settings**.